src/couch_index/src/couch_index_compactor.erl

```diff
@@ -81,6 +85,14 @@ handle_cast(_Mesg, State) ->
 
 handle_info({'EXIT', Pid, normal}, #st{pid=Pid}=State) ->
     {noreply, State#st{pid=undefined}};
+handle_info({'EXIT', Pid, Reason}, #st{pid = Pid} = State) ->
+    #st{idx = Idx, mod = Mod} = State,
+    {ok, IdxState} = gen_server:call(Idx, {compaction_failed, Reason}),
```

Comment:
Since there seems to be a disagreement, should we try to find a tiebreaker?

In any case, I'm unable to merge this PR, so a committer will need to do the honors.